clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -450,13 +453,15 @@ static void checkSYCLType(SemaSYCL &S, QualType Ty, SourceRange Loc,
   while (Ty->isAnyPointerType() || Ty->isArrayType())
     Ty = QualType{Ty->getPointeeOrArrayElementType(), 0};
 
-  // __int128, __int128_t, __uint128_t, long double, __float128
-  if (Ty->isSpecificBuiltinType(BuiltinType::Int128) ||
-      Ty->isSpecificBuiltinType(BuiltinType::UInt128) ||
-      Ty->isSpecificBuiltinType(BuiltinType::LongDouble) ||
-      Ty->isSpecificBuiltinType(BuiltinType::BFloat16) ||
-      (Ty->isSpecificBuiltinType(BuiltinType::Float128) &&
-       !S.getASTContext().getTargetInfo().hasFloat128Type())) {
+  if (((Ty->isFloat128Type() ||
```

Comment:
I wonder if it makes sense instead of improving SemaSYCL routine, reuse and if needed improve `checkTypeSupport` routine that already exists in upsteam?

Reply:
I agree there is something to do here, especially for upstreaming. But I don't have the bandwidth right now to investigate / refactor / test. Do you mind I create an issue and tackle this later on ? That'll unblock the team needing this change.

Reply:
Sure, no problem